x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/relevance_tuning_logic.test.ts

```diff
@@ -0,0 +1,296 @@
+
+        expect(RelevanceTuningLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          searchSettings,
+          unsavedChanges: true,
+        });
+      });
+    });
+
+    describe('setFilterValue', () => {
+      it('should set state', () => {
+        mount();
+        RelevanceTuningLogic.actions.setFilterValue('foo');
+
+        expect(RelevanceTuningLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          filterInputValue: 'foo',
+        });
+      });
+    });
+
+    describe('setSearchQuery', () => {
+      it('should set state', () => {
+        mount();
+        RelevanceTuningLogic.actions.setSearchQuery('foo');
+
+        expect(RelevanceTuningLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          query: 'foo',
+        });
+      });
+    });
+
+    describe('setSearchResults', () => {
+      it('should set state', () => {
+        mount({
+          resultsLoading: true,
+        });
+        RelevanceTuningLogic.actions.setSearchResults(searchResults);
+
+        expect(RelevanceTuningLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          searchResults,
+          resultsLoading: false,
+        });
+      });
+    });
+
+    describe('setResultsLoading', () => {
+      it('should set state', () => {
+        mount({
+          resultsLoading: false,
+        });
+        RelevanceTuningLogic.actions.setResultsLoading(true);
+
+        expect(RelevanceTuningLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          resultsLoading: true,
+        });
+      });
+    });
+
+    describe('clearSearchResults', () => {
+      it('should set state', () => {
+        mount({
+          searchResults: [{}],
+        });
+        RelevanceTuningLogic.actions.clearSearchResults();
+
+        expect(RelevanceTuningLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          searchResults: null,
+        });
+      });
+    });
+
+    describe('resetSearchSettingsState', () => {
+      it('should set state', () => {
+        mount({
+          dataLoading: false,
+        });
+        RelevanceTuningLogic.actions.resetSearchSettingsState();
+
+        expect(RelevanceTuningLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          dataLoading: true,
+        });
+      });
+    });
+
+    describe('dismissSchemaConflictCallout', () => {
+      it('should set state', () => {
+        mount({
+          showSchemaConflictCallout: true,
+        });
+        RelevanceTuningLogic.actions.dismissSchemaConflictCallout();
+
+        expect(RelevanceTuningLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          showSchemaConflictCallout: false,
+        });
+      });
+    });
+  });
+
+  describe('selectors', () => {
+    describe('engineHasSchemaFields', () => {
+      it('should false if there is only a single field in a schema', () => {
+        // I believe this is because schemas *always* have an ID, and we don't consider that
+        // a tunable field
```

Comment:
We should validate this with another team member and remove the ambiguity from this comment

Reply:
True. I updated the comment: 62e9598c1f2b21bfcbd01c8954c0e0d274c50b5b